eve/io/mongo/mongo.py

```diff
@@ -736,15 +737,19 @@ def bulk_update(self, resource, lookup_fields, doc_or_docs):
                 # perform bulk write operations in chunks
                 if len(operations) == 1000:
                     coll.bulk_write(operations, ordered=True)
-                    ids.extend([resp['_id'] for resp in coll.find(bulk_lookup_queries, projection=["_id"])])
+                    for resp in coll.find(bulk_lookup_queries):
+                        ids.append(resp['_id'])
+                        response_documents.append(resp)
```

Comment:
I am curious why we are returning IDs as separate list when they are already present in "resp" which is being returned as well.

Reply:
We are returning "ids" and "response_documents". The documents does already include the ID but in the  eve/methods/post.py both lists are needed to build the response (unless there is a custom "id_field" that defined in the resource). https://github.com/catalogicsoftware/eve/blob/catalogic-1.1.5/eve/methods/post.py#L271-L286

Reply:
Wouldn't it be possible to construct ID list from response document list where ever it is needed? I hope I am not missing something here.

Reply:
yes we can do that. The bulk_update could just return the list of documents and in the post.py it could check if the _id field exists. That would be better. I will make the change.